### PR TITLE
[Refine] zero_param_level_sharding handling for fewer parameters

### PR DESCRIPTION
**Pull request by 0xWJ**
Enhance support for scenarios where zero_param_level_sharding is enabled but the number of parameters is less than expected. This includes adding padding to ensure proper functionality and updating related tests for correctness.

### CODE REVIEW

File: tests/cli/test_trainer_muon.py [hunk 1]
```diff
@@ -282,3 +282,37 @@ def param_clss_fn2(param_name: str) -> tuple[int, int]:
...
+    else:
+        return 0, 0, {'zero_param_level_sharding': None}
+
+
+
+@pytest.mark.skipif(not torch.cuda.is_available() or torch.cuda.device_count() < 2, reason='lack of gpu devices')
+def test_trainer_muon_resume_correctness_zero1_param_config_less_params(tmp_path):
+    config_file = 'trainer_args_muon_hybrid.yaml'
+
+    launch_torchrun(2, trainer_muon_worker, tmp_path, config_file, '2', [
```
[the remark this answers is not in the document]
Comment (0xWJ) [reply tests/cli/test_trainer_muon.py]: fixed

File: nnscaler/runtime/muon_optimizer.py [hunk 2]
```diff
@@ -63,23 +63,25 @@ def state_dict(self):
...
-                    ]
-                    new_param_states[flat_idx] = {self.momentum_buffer_name: fpi.flatten(embeded_states, device='cpu')}
-                #else: no state for this flattened param
-
-            state['state'] = new_param_states
+        # But unfortunately,
+        # we can't distinguish between above case and normal case
+        # (when there is no params in optimizer or all params are not used in any step )
+        # so we need to handle the empty state case always.
+        new_param_states = {}
```
[the remark this answers is not in the document]
Comment (0xWJ) [reply nnscaler/runtime/muon_optimizer.py:70]: no fix

File: nnscaler/runtime/adapter/reducer.py [hunk 3]
```diff
@@ -137,13 +137,15 @@ def flatten(self, tensors: list[Optional[torch.Tensor]], *, device=None) -> torc
             device: the device of the result flattened tensor,
                 if None, use the device of the first non-None tensor
```
[the remark this answers is not in the document]
Comment (0xWJ) [reply nnscaler/runtime/adapter/reducer.py]: fixed.

File: nnscaler/runtime/muon_optimizer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lynex) [reply nnscaler/runtime/muon_optimizer.py:70]: Shall we fill an issue for future fix?

**Review by lynex — COMMENTED** [review]
I recommend filing an issue regarding empty states being instantiated as tensor chunks, leading to additional and unnecessary storage usage.

**Comment by 0xWJ** [comment]
> I recommend filing an issue regarding empty states being instantiated as tensor chunks, leading to additional and unnecessary storage usage.

i think this is not important. This only happens once when a hybridoptimizer is instantiated.  And the storage is allocated in cpu, not GPU.

**Review by lynex — APPROVED** [review]
